game/src/main/scala/io/aigar/game/Resources.scala

```diff
@@ -45,8 +46,27 @@ class ResourceType(grid:Grid, val min: Int, val max: Int, mass: Int, score: Int)
 
   def spawnResources: Unit = {
     val ratio = (positions.length - min).toFloat / (max - min)
-    if (scala.util.Random.nextFloat >= ratio) {
+    if ((scala.util.Random.nextFloat + 0.000001) >= ratio) {
```

Comment:
Could you explain why this is required? Would http://www.scala-lang.org/api/2.7.2/scala/Math$object.html#EPS_FLOAT be more apporpiated?